### PR TITLE
added subtract operation

### PR DESCRIPTION
**Pull request by mungu42**

### CODE REVIEW

File: src/rdd/rdd.rs [hunk 1]
```diff
@@ -26,28 +26,49 @@ use serde_derive::{Deserialize, Serialize};
 use serde_traitobject::{Deserialize, Serialize};
 
 mod parallel_collection_rdd;
+
 pub use parallel_collection_rdd::*;
```
Comment (rajasekarv) [inline src/rdd/rdd.rs:30]: can you remove these extra new lines?

File: src/rdd/rdd.rs [hunk 2]
```diff
@@ -838,6 +859,65 @@ pub trait Rdd: RddBase + 'static {
         self.intersection_with_num_partitions(other, self.number_of_splits())
     }
 
+    fn subtract<T>(&self, other: Arc<T>) -> SerArc<dyn Rdd<Item = Self::Item>>
```
Comment (rajasekarv) [inline src/rdd/rdd.rs:862]: If possible, please add documentation for this method. Similar to the one in Spark will do.

File: src/rdd/rdd.rs [hunk 3]
```diff
@@ -838,6 +859,65 @@ pub trait Rdd: RddBase + 'static {
...
+                if (v1.len() >= 1) ^ (v2.len() >= 1) {
+                    Some(x)
+                } else {
+                    None
+                }
+            })))
+            .map_partitions(Box::new(Fn!(|iter: Box<
+                dyn Iterator<Item = Option<Self::Item>>,
+            >|
+             -> Box<
```
Comment (rajasekarv) [inline src/rdd/rdd.rs:909]: I maybe wrong, but it looks like rustfmt is not done. If so, please run cargo fmt.

File: src/rdd/rdd.rs (hunk 1 above)
Comment (mungu42) [reply src/rdd/rdd.rs]: removed

File: src/rdd/rdd.rs (hunk 2 above)
Comment (mungu42) [reply src/rdd/rdd.rs:843]: have added documentation for the function and an example in examples/subtract.rs

File: src/rdd/rdd.rs (hunk 3 above)
Comment (mungu42) [reply src/rdd/rdd.rs:889]: done

**Comment by iduartgomez** [comment]
Closes #91 

There is an optimized version for this op (could be implemented in the future) in case you want to check it out. It entails quite more changes though, I had somethings close to done (had to debug one error) but haven't been able to finish it up yet due to lack of time.

**Comment by mungu42** [comment]
Is there any reference to the more optimized method you mentioned, I could look into
?

File: tests/test_rdd.rs [hunk 4]
```diff
@@ -670,3 +670,16 @@ fn test_take_ordered() {
...
+fn test_subtract() {
+    let sc = CONTEXT.clone();
+    let col1 = vec![1, 2, 3, 4, 5, 10, 12, 13, 19, 0];
+
+    let col2 = vec![3, 4, 5, 6, 7, 8, 11, 13];
+
+    let first = sc.parallelize(col1, 4);
+    let second = sc.parallelize(col2, 4);
+    let ans = first.subtract(Arc::new(second));
+    assert_eq!(ans.collect().unwrap(), vec![19, 12, 10, 1, 0, 2])
```
Comment (rajasekarv) [inline tests/test_rdd.rs:684]: This assertion can fail randomly depending on shuffle(as evidenced by CI check fail). You can iterate over one vector and check if an element exists in the other one or convert both of them as hashset and do assert_eq. HashSet is the better way.

**Comment by rajasekarv** [comment]
merging this for now as it is working version. @iduartgomez When you are done, please feel free to modify this operation.

**Pull request merged**